src/wrappers/web3.js

```diff
@@ -1,154 +1,92 @@
+export let Web3Loaded = Web3LoadedDeferred.promise
+export let TBTCLoaded = TBTCLoadedDeferred.promise
 
-                this.setState({
-                    balance,
-                    account: accounts[0]
-                })
-            }
-        }
-    }
+const initializeContracts = async (web3, connector) => {
+    // Initialise default account.
+    const accounts = await web3.eth.getAccounts()
+    web3.eth.defaultAccount = accounts[0]
 
-    getBalanceForAccount = async (account) => {
-        const { web3 } = this.state
+    // Log the netId/chainId.
+    const netId = await web3.eth.net.getId()
+    const chainId = await web3.eth.getChainId()
+    console.debug(`netId: ${netId}\nchainId: ${chainId}`)
 
-        if (web3) {
-            return await web3.eth.getBalance(account)
-        }
-    }
-
-    render() {
-        const { account, balance, loading, web3 } = this.state
-
-        const contextValue = {
-            account,
-            balance,
-            loading,
-            web3,
-            connectDapp: this.connectDapp
-        }
+    Web3LoadedDeferred.resolve(web3)
+    
+    const tbtc = await TBTC.withConfig({
+        web3: web3,
+        bitcoinNetwork: "testnet",
+        electrum: config.electrum
+    })
 
-        return (
-            <Web3Context.Provider value={contextValue}>
-                {this.props.children}
-            </Web3Context.Provider>
-        )
-    }
+    TBTCLoadedDeferred.resolve(tbtc)
 }
 
-function withWeb3(Child) {
-    return (props) => (
-        <Web3Context.Consumer>
-            {({ web3 }) => <Child {...props} web3={web3} />}
-        </Web3Context.Consumer>
-    )
+function getLibrary(provider, connector) {
+    const library = new Web3(provider)
+    return library
 }
 
-function withAccount(Child) {
-    return (props) => (
-        <Web3Context.Consumer>
-            {({ account }) => <Child {...props} account={account} />}
-        </Web3Context.Consumer>
-    )
-}
+const Web3ReactManager = ({ children }) => {
+    const { activate, active, library, connector } = useWeb3React()
+
+    useEffect(() => {
+        if(active) {
+            initializeContracts(library, connector)
+        }
+    }, [active])
+
+    // Watch for changes:
+    // provider = this.state.web3.eth.currentProvider
+    // provider.on('networkChanged', this.getAndSetAccountInfo)
+    // provider.on('accountsChanged', this.getAndSetAccountInfo)
 
-function withBalance(Child) {
-    return (props) => (
-        <Web3Context.Consumer>
-            {({ balance }) => <Child {...props} balance={balance} />}
-        </Web3Context.Consumer>
-    )
+    return children
 }
 
-function withConnectDapp(Child) {
-    return (props) => (
-        <Web3Context.Consumer>
-            {({ connectDapp, loading }) => <Child {...props} connectDapp={connectDapp} loading={loading} />}
-        </Web3Context.Consumer>
-    )
+const Web3Wrapper = ({ children }) => {
+    return <Web3ReactProvider getLibrary={getLibrary}>
```

Comment:
Why `Library`?

Reply:
Que?

Reply:
Why the term `Library`? It seems out of place here. You're just getting a Web3 instance, right? Library in my mind is a bundle of code; offhand if I saw this I would expect it to return something like one of several libraries that could be used.

I just realized though, the property name is from web3-react (these are the things you miss when it's the end of the day <_<). Still, I wouldn't mind if our side was named something like `buildWeb3` or `instantiateWeb3` or similar.

Reply:
Ah, I see. Yeah it originates from the web3-react package supporting ethers or web3.js. Will change.